test/unit/black-button_test.js

```diff
@@ -184,40 +353,138 @@ describe('Black button testing', function() {
-                json: utils.readExampleFile('./test/unit/provision/synchronousButtonProvision.json'),
-                headers: {
-                    'fiware-service': 'smartGondor',
-                    'fiware-servicepath': '/gardens'
+                form: {
+                    cadena: '#STACK1#0,BT,S,6,FFE876AE,0$'
                 }
-            };
+            },
+            originalGenerateInternalId;
 
-            request(registerOptions, function(error, response, body) {
+        beforeEach(function(done) {
+            var request = './test/unit/contextRequests/blackButtonSynchronousRequest.json',
+                response = './test/unit/contextResponses/blackButtonSynchronousStatusCode500.json';
+
+            config.ngsi.plainFormat = true;
+
+            originalGenerateInternalId = idGenerator.generateInternalId;
+            idGenerator.generateInternalId = mockedGenerateInternalId;
+
+            utils.contextBrokerMock.push(nock('http://' + config.ngsi.contextBroker.host + ':1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext', utils.readExampleFile(request))
+                .reply(function(uri, requestBody, cb) {
+                    setTimeout(sendUpdateLazyAttributes, 500);
+
+                    cb(null, [200, utils.readExampleFile(response)]);
+                }));
+
+            utils.contextBrokerMock.push(nock('http://' + config.ngsi.contextBroker.host + ':1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/NGSI9/registerContext')
+                .reply(200,
+                utils.readExampleFile('./test/unit/contextAvailabilityResponses/registerDeviceSuccess.json')));
+
+            registerDevice(done);
+        });
+
+        afterEach(function(done) {
+            config.ngsi.plainFormat = false;
+
+            idGenerator.generateInternalId = originalGenerateInternalId;
+
+            utils.contextBrokerMock.push(nock('http://' + config.ngsi.contextBroker.host + ':1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/NGSI9/registerContext')
+                .reply(200,
+                utils.readExampleFile('./test/unit/contextAvailabilityResponses/registerDeviceSuccess.json')));
+
+            iotagentNodeLib.unregister('STACK1', done);
+        });
+
+        it('should return an explanation the appropriate error code', function(done) {
+            request(options, function(error, result, body) {
                 should.not.exist(error);
-                response.statusCode.should.equal(200);
-                callback();
+                result.statusCode.should.equal(200);
+                body.should.equal('#STACK1#0,BT,S,0,0:500,rgb-66CC00;t-2,0$');
```

Comment:
Since the code `rgb-66CC00;t-2` is part of the corresponding `.json` file used in this test case, what about including it as `require('xxx.json').contextResponses.statusCode.reasonPhrase` instead of as hard-coded text in the test case assertions (of course, the `require()` could be done in the 'beforeEach()' function (if not already done) or any place you consider appropriate... :)